cartography/driftdetect/detect_drift.py

```diff
@@ -1,127 +1,97 @@
+import os
 
+from cartography.driftdetect.model import load_state_from_json_file
+from cartography.driftdetect.report_info import load_report_info_from_json_file
 
 logger = logging.getLogger(__name__)
 
 
-def perform_drift_detection(session, expect_folder, update):
+def perform_drift_detection(query_directory, start_state_file, end_state_file):
     """
-    Perform Drift Detection
-    :type session: neo4j_session
-    :param session: session with infrastructure information
-    :type expect_folder: string
-    :param expect_folder: detector directory
-    :type update: Decides whether or not to update the graph
-    :param update: boolean
-    :return: list of tuples of drift dictionaries and detectors
+    Performs Drift Detection.
+
+    :type query_directory: String
+    :param query_directory: Path to the query directory.
+    :type start_state_file: String
+    :param start_state_file: The filename of the earlier state chronologically to be compared to.
+    :type end_state_file: String
+    :param end_state_file: The filename of the later state chronologically to be compared to.
+    :return:
     """
-    drift_info_detector_pairs = []
-    for root, _, filenames in os.walk(expect_folder):
-        for filename in filenames:
-            file_path = os.path.join(root, filename)
-            try:
-                detector = load_detector_from_json_file(file_path)
-                for drift_info in detector.run(session, update):
-                    drift_info_detector_pairs.append((drift_info, detector))
-                if update:
-                    try:
-                        write_detector_to_json_file(detector, file_path)
-                    except ValidationError as err:
-                        msg = "Unable to save DriftDetector from file {0}.\n{1}".format(file_path, err.messages)
-                        logger.exception(msg)
-            except ValidationError as err:
-                msg = "Unable to create DriftDetector from file {0}.\n{1}".format(file_path, err.messages)
-                logger.exception(msg)
-    return drift_info_detector_pairs
+    report_info = load_report_info_from_json_file(os.path.join(query_directory, "report_info.json"))
+    start_state = load_state_from_json_file(os.path.join(query_directory, report_info.shortcuts.get(
+        start_state_file,
+        start_state_file)))
+    end_state = load_state_from_json_file(os.path.join(query_directory, report_info.shortcuts.get(
+        end_state_file,
+        end_state_file)))
+    if start_state.name != end_state.name or start_state.validation_query != end_state.validation_query:
+        raise Exception("Query Information does not match.")
+    new_results, missing_results = compare_states(start_state, end_state)
+    return new_results, missing_results
+
 
+def compare_states(start_state, end_state):
+    """
+    Compares drift between two DriftStates.
 
-def run(config):
+    :type start_state: DriftState
```

Comment:
I was thinking about changing the name to state instead of driftstate and just getting rid of the word "drift" altogether in the code documentation, but initially I wanted to add the "drift" part to reference the service name. However, drift isn't a very self explanatory word and kind of obfuscates the intent so I agree with getting rid of it.